### PR TITLE
Deprecate Python 3.7 and earlier, but allow cqlsh to run with Python 3.6-3.11

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe; reviewed by ? for CASSANDRA-19467

### CODE REVIEW

File: .build/run-python-dtests.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ if [ "${java_version}" -eq 17 ] && [[ "${target}" == "dtest-upgrade" ]] ; then
 fi
 
 python_version=$(python -V 2>&1 | awk '{print $2}' | awk -F'.' '{print $1"."$2}')
-python_regx_supported_versions="^(3.8|3.9|3.10|3.11)$"
+python_regx_supported_versions="^(3.6|3.7|3.8|3.9|3.10|3.11)$"
```
Comment (michaelsembwever) [inline .build/run-python-dtests.sh]: the docker images also need to be updated for this to work. 
https://github.com/apache/cassandra/blob/cassandra-5.0/.build/docker/ubuntu2004_test.docker

Comment (michaelsembwever) [reply .build/run-python-dtests.sh]: and…  this script has nothing to do with the cqlsh tests :)

Comment (maedhroz) [reply .build/run-python-dtests.sh]: I think this bit just needs to be reverted.

Comment (maedhroz) [reply .build/run-python-dtests.sh]: Done

**Comment by maedhroz** [comment]
Trunk patch at https://github.com/apache/cassandra/pull/3183

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/8a33f323508dab061dbebfd168e431cd0e683e01

**Pull request closed without merging**